apps/web/lib/team/[slug]/getServerSideProps.tsx

```diff
@@ -175,6 +178,28 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
     } as const;
   }
 
+  const organizationSettings = isOrgContext
+    ? team?.isOrganization
+      ? team?.organizationSettings
+      : team?.parent?.organizationSettings
+    : null;
+  let allowSEOIndexing = false;
+
+  if (!!organizationSettings) {
+    allowSEOIndexing = organizationSettings.allowSEOIndexing;
+
+    if (organizationSettings.disableOrgSubdomainURL) {
+      const redirect = getMainDomainOrgRedirect(
+        context.req,
+        team?.children?.map((team) => team.slug || "").filter((slug) => !!slug),
```

Comment:
Why do we need team.children? The requirement is to do the redirect only for organization profile.  It isn't applicable to Team Profile.


The motivation for the issue this PR is fixing is that people shouldn't be able to see a list of all the teams/users of the organization. That info is shown on organization profile page. So, the best was to redirect acme.cal.com to acme.com itself.

Reply:
I think we could simplify the logic to just redirect to orgAutoAcceptEmail.

We don't need `getMainDomainOrgRedirect` function.

`redirectToVerifiedDomain({verifiedDomain})`

Function defined in this module.
```js
function getRedirectToVerifiedDomain({ verifiedDomain }: { verifiedDomain: string }) {
  return {
    redirect: {
      permanent: true,
      destination: `https://${verifiedDomain}`,
    },
  };
}
```

We know that this route is used in 3 cases. We just need to ensure that the function is called only in the 3rd case.
1. Team Profile Page
2. SubTeam Profile
3. Organization Profile Page

Reply:
the same page i.e. team/[slug] is used for both org home page as well as for org/[team_slug] like (org/marketing).
here team in team?.children in org context is nothing but org.children (or teams under org).

so team?.children was used to validate team slug `acme.cal.com/[marketing]`

So do you mean, we need to redirect only when home page is visited `acme.cal.com` and no need for 
`acme.cal.com/marketing`

Reply:
Yeah absolutely. See the updated description of the issue
![image](https://github.com/user-attachments/assets/98ebfa21-5aa2-4a0f-b166-6042b438d116)

Reply:
got it